src/mvba/tag.rs

```diff
@@ -0,0 +1,42 @@
+use std::fmt::Display;
+
+use super::NodeId;
+
+#[derive(Debug, Clone, Eq, PartialEq, serde::Serialize, serde::Deserialize)]
+pub struct Domain {
+    pub id: String, // this is same as $ID$ in spec
+    pub seq: usize, // this is same as $s$ in spec
+}
+
+impl Domain {
+    pub fn new(id: impl Into<String>, seq: usize) -> Self {
```

Comment:
Consider `ID` as string. Owner of the consensus can define `Domain` that can be convert into string.

Reply:
Not sure I'm getting your suggestions, are you suggesting we replace the `id: impl Into<String>` with `id: String`?

I did it this way so that we can write `Domain::new("membership", 0)` instead of `Domain::new("membership".to_string(), 0)`.

Reply:
I mean  
` pub fn new(id: &str, seq: usize) -> Self { ... }`

Reply:
Then you force the implementation to clone(), let's stick with the `impl Into<String>`